cleanup-spurious-images.py

```diff
@@ -131,21 +137,36 @@ def cluster_connect(pool, conffile, rados_id):
     ioctx = cluster_connect(cfg.pool, cfg.conf, cfg.user)
     rbd_inst = rbd.RBD()
     sess = make_session(cfg)
-    cclient = cinder_client.Client('2', session=sess)
+    cclient = cinder_client.Client('2', session=sess,region_name=cfg.os_region_name)
+
+    vols_snaps = {}
+
+    for vol in rbd_inst.list(ioctx):
+        if volume_re.match(vol):
+            image = rbd.Image(ioctx,vol)
+            vols_snaps[vol]=rbd.Image.list_snaps(image)
 
-    volumenames = [vol for vol in rbd_inst.list(ioctx) if volume_re.match(vol)]
-    log.info("Got information about %d volumes", len(volumenames))
-    # Inizializza una lista
     to_delete= []
-    for name in volumenames:
-        uuid = volume_re.search(name).group('uuid')
+    for vol in vols_snaps:
+        uuid = volume_re.search(vol).group('uuid')
```

Comment:
This is already checked in the for loop above.

There are rbd volumes that match <volume>-uuid created by cinder and other <uuid>_disk created by nova.
The goal is to detect rbd volumes that match these patterns and do not exist in the Openstack database.